### PR TITLE
Feature/pkg build process

### PR DESCRIPTION
**Pull request by gaborcsardi**
Uses https://github.com/r-lib/withr/pull/68, so it would make sense to merge that first.

### CODE REVIEW

File: tests/testthat/test-build-process.r [hunk 1]
```diff
@@ -0,0 +1,92 @@
+context("rcmdbuild_process")
+
+# Package without source code --------------------------------------------
+
+test_that("source builds return correct filenames", {
```
[the remark this answers is not in the document]
Comment (gaborcsardi) [reply tests/testthat/test-build-process.r]: The difficulty with that is that there is a timing issue. If the build finished before I issue the kill, then I cannot kill it. I guess I can still make it robust for this, and skip in this case. Let me try.

File: R/build-bg.R [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  public = list(
+
+    initialize = function(path = ".", dest_path = NULL, binary = FALSE,
+                          vignettes = TRUE, manual = FALSE, args = NULL)
+      rcb_init(self, private, super, path, dest_path, binary, vignettes,
+               manual, args),
+
+    get_dest_path = function() private$dest_path,
+
+    kill = function(...) {
```
[the remark this answers is not in the document]
Comment (gaborcsardi) [reply R/build-bg.R]: Yeah, good idea.

File: tests/testthat/test-build-process.r (hunk 1 above)
[the remark this answers is not in the document]
Comment (gaborcsardi) [reply tests/testthat/test-build-process.r]: What is a good way to delay a build for a couple of seconds? configure / configure.win with some sleeping?

[the remark this answers is not in the document]
Comment (gaborcsardi) [reply tests/testthat/test-build-process.r]: So, the finalizer will run not when the process finishes, but when the object is garbage collected. But even that is better than nothing.

File: R/build-bg.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (jimhester) [reply R/build-bg.R]: Yeah, that is what I expected and is fine I think

**Comment by gaborcsardi** [comment]
OK, I added a finalizer, and also a test for killing. 

Killing is inherited from processx/callr, and it is heavily tested there, so I am not too worried about it.

**Comment by gaborcsardi** [comment]
Btw. wait for the windows CI, because the exit code test is windows specific.

**Comment by gaborcsardi** [comment]
I rebased, this might solve the CI failures....

**Comment by jimhester** [comment]
Thanks!

**Pull request merged**